cocotb_build_libs.py

```diff
@@ -235,7 +219,7 @@ def _get_common_lib_ext(include_dir, share_lib_dir):
         libraries=[_get_python_lib_link(), "gpilog", "cocotbutils"],
         library_dirs=python_lib_dirs,
         sources=[os.path.join(share_lib_dir, "embed", "gpi_embed.cpp")],
-        extra_link_args=_extra_link_args(lib_name="libcocotb", rpath="$ORIGIN"),
+        extra_link_args=_extra_link_args(lib_name="libcocotb", rpaths=["$ORIGIN"] + python_lib_dirs),
```

Comment:
So we have to add `python_lib_dirs` here because you get an error 
```
dyld: Library not loaded: @rpath/libpython3.7m.dylib
  Referenced from: .../lib/python3.7/site-packages/cocotb/libs/libcocotb.so
  Reason: image not found
```
Why does it work when using `brew`?

Reply:
brew python is always installed in a fixed global location. So the `install_name` of `libpython` contains the full path and any library that links to `libpython` will look for it in the full path. So, rpath is not needed. Compare the output of `otool -L libcocotb.so` (Mac-equivalent of `ldd`) on the master branch for brew vs conda pythons:

For brew python:
```
libcocotb.so:
        @rpath/libcocotb.so (compatibility version 0.0.0, current version 0.0.0)
        /usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/Python (compatibility version 3.7.0, current version 3.7.0)
        @rpath/libgpilog.so (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libcocotbutils.so (compatibility version 0.0.0, current version 0.0.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
```

For conda python:
```
libcocotb.so:
        @rpath/libcocotb.so (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libpython3.7m.dylib (compatibility version 3.7.0, current version 3.7.0)
        @rpath/libgpilog.so (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libcocotbutils.so (compatibility version 0.0.0, current version 0.0.0)
        @rpath/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
```

Hence, `rpath` needs `python_lib_dirs` for conda python.

Reply:
Thank you for your reply.
It seems adds up complication if we ever would want to distribute binary package (as you commented before) but this is one of many issues ...